### ISSUE TITLE
async function declaration gets stripped

### ISSUE THREAD

**Issue by catch56** [mck89/peast#58]
There seems to be an issue parsing async function declarations, couldn't find an existing or old issue about this.

Given:

```
Foo = {
  async bar() {},
  baz: async () => {},
}
```
Run through:

```
$ast = Peast::latest($data)->parse();
$renderer = new Renderer();
$renderer->setFormatter(new CompactFormatter());
print $renderer->render($ast);
```

You would expect:
```
Foo={async bar(){},baz:async()=>{}};
```

But instead you get:
```
Foo={bar(){},baz:async()=>{}};
```

Steps to reproduce aren't mine, just copying from: https://www.drupal.org/project/drupal/issues/3374660

**Comment by Niklan** [comment]
We faced this on the project with AlpineJS. Basically, code was something like:

```javascript
Alpine.data('foo', () => ({
  async bar() {
    await Baz.bar();
  }
}))
```

This worked fine until Drupal 10.1 update, which introduced Peast as JS optimizer.

We updated our code to be:

```javascript
Alpine.data('foo', () => ({
  bar: async () => {
    await Baz.bar();
  }
}))
```

This solves the problem, but seems like a bug, because the first example is also a valid JS syntax. Peast removes async from `bar()` and browser fails with `await` notation without `async` for function.
